test/parallel/test-source-map-enable.js

```diff
@@ -283,6 +283,24 @@ function nextdir() {
   assert.ok(output.stderr.toString().includes('webpack:///webpack.js:14:9'));
 }
 
+// Stores and applies source map associated with file that throws while
+// being required.
+{
+  const coverageDirectory = nextdir();
```

Comment:
@ljharb @aduh95 @benjamingr, this test points out the motivation for this refactor. If an exception was thrown during a `require`, we'd lose the ability to remap the stack trace using a source map in the module: